app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationRouter.kt

```diff
@@ -47,150 +51,119 @@ import com.wire.android.ui.common.visbility.rememberVisibilityState
 import com.wire.android.ui.destinations.ConversationScreenDestination
 import com.wire.android.ui.destinations.NewConversationSearchPeopleScreenDestination
 import com.wire.android.ui.destinations.OtherUserProfileScreenDestination
-import com.wire.android.ui.home.HomeSnackbarState
 import com.wire.android.ui.home.conversations.PermissionPermanentlyDeniedDialogState
 import com.wire.android.ui.home.conversations.details.dialog.ClearConversationContentDialog
 import com.wire.android.ui.home.conversations.details.menu.DeleteConversationGroupDialog
 import com.wire.android.ui.home.conversations.details.menu.LeaveConversationGroupDialog
 import com.wire.android.ui.home.conversationslist.all.AllConversationScreenContent
-import com.wire.android.ui.home.conversationslist.call.CallsScreenContent
-import com.wire.android.ui.home.conversationslist.mention.MentionScreenContent
 import com.wire.android.ui.home.conversationslist.model.ConversationItem
 import com.wire.android.ui.home.conversationslist.model.ConversationsSource
 import com.wire.android.ui.home.conversationslist.model.DialogState
 import com.wire.android.ui.home.conversationslist.model.GroupDialogState
 import com.wire.android.ui.home.conversationslist.model.isArchive
 import com.wire.android.ui.home.conversationslist.search.SearchConversationScreen
 import com.wire.android.util.permission.PermissionDenialType
+import com.wire.android.util.ui.SnackBarMessageHandler
 import com.wire.kalium.logic.data.id.ConversationId
 import com.wire.kalium.logic.data.user.UserId
-import kotlinx.coroutines.flow.MutableSharedFlow
 
 // Since the HomeScreen is responsible for displaying the bottom sheet content,
 // we create a bridge that passes the content of the BottomSheet
 // also we expose the lambda which expands the BottomSheet from the HomeScreen
+@OptIn(ExperimentalMaterial3Api::class)
 @Suppress("ComplexMethod")
 @Composable
 fun ConversationRouterHomeBridge(
     navigator: Navigator,
     conversationItemType: ConversationItemType,
-    onHomeBottomSheetContentChanged: (@Composable ColumnScope.() -> Unit) -> Unit,
-    onOpenBottomSheet: () -> Unit,
-    onCloseBottomSheet: () -> Unit,
-    onSnackBarStateChanged: (HomeSnackbarState) -> Unit,
     searchBarState: SearchBarState,
-    isBottomSheetVisible: () -> Boolean,
-    conversationsSource: ConversationsSource = ConversationsSource.MAIN
+    conversationsSource: ConversationsSource = ConversationsSource.MAIN,
+    conversationListViewModel: ConversationListViewModel = hiltViewModel(),
+    conversationCallListViewModel: ConversationCallListViewModel = hiltViewModel(),
 ) {
+    var currentConversationItem by remember {
```

Comment:
suggestion: maybe `currentSheetConversationItem` to emphasise that this mutable state is used for the bottom sheet?